sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -236,6 +236,11 @@ private void drainQueue() {
                 numberRequested = REQUESTED.addAndGet(this, -numberConsumed);
             }
         }
+        if (numberRequested == 0L) {
+            logger.atVerbose()
+                .log("Current work is completed. Schedule next work.");
+            getOrUpdateCurrentWork();
```

Comment:
I assume after `getOrUpdateCurrentWork` and next work in queue become current work, and requested updated, this Subscriber will need to get into the `drain/drainQueue` loop again? Do we know whether this is the case, and where in code this happens?

Reply:
>after getOrUpdateCurrentWork called, next work in queue become current work, and requested number updated

This is right. But `numberRequested==0L` block is not in loop, so `drain/drainQueue` return after requested number updated. Only when next message received, it will trigger this `drain/drainQueue` method again.

Reply:
The `hookOnNext` method?

Reply:
yes, the receive flow: ondelivery -> hookOnNext -> drain -> drainQueue